fvm/src/machine/engine.rs

```diff
@@ -71,9 +81,32 @@ impl Default for MultiEngine {
     }
 }
 
-pub fn default_wasmtime_config() -> wasmtime::Config {
+pub fn default_wasmtime_config(
+    instance_count: u32,
+    instance_memory_maximum_size: u64,
+) -> wasmtime::Config {
     let mut c = wasmtime::Config::default();
 
+    // wasmtime default: OnDeand
+    // We want to pre-allocate all permissible memory to support the maximum allowed recursion limit.
+    c.allocation_strategy(InstanceAllocationStrategy::Pooling {
+        strategy: PoolingAllocationStrategy::ReuseAffinity,
+        instance_limits: InstanceLimits {
+            count: instance_count,
+            // Adjust the maximum amount of host memory that can be committed to an instance to
+            // match the static linear memory size we reserve for each slot.
+            memory_pages: instance_memory_maximum_size / WASM_PAGE_SIZE,
```

Comment:
nit: we should probably just ask the user to specify the number of pages (defined to be 16KiB per wasm spec).

Reply:
I wanted to stick to bytes because that seems to be the way in `wasmtime::Config`. If you look at [static_memory_maximum_size](https://github.com/bytecodealliance/wasmtime/blob/main/crates/wasmtime/src/config.rs#L1109) it also does this division. 

It seems better to not require the user to possess extra knowledge about the wasm spec. Especially since it seems like if 16KB is in the spec then this value is [different](https://github.com/bytecodealliance/wasmtime/blob/main/crates/environ/src/lib.rs#L65-L66). 

The docs of `memory_pages` say the default value of 160 means 10MB, which works out with 160*65,536=10,485,760. `pooling.rs` will [use WASM_PAGE_SIZE to limit memory](https://github.com/bytecodealliance/wasmtime/blob/main/crates/runtime/src/instance/allocator/pooling.rs#L746), so I think it's more natural to hide this from the user and just ask in terms of bytes.

Reply:
My concern is rounding. Alternatively, https://github.com/filecoin-project/ref-fvm/pull/1073

Reply:
I sort of swept it under my mental carpet by thinking any remainder will be truncated and we'll have something that will work. Or that we can add some validation somewhere to ensure it's a multiple of the page size.